matminer/featurizers/base.py

```diff
@@ -364,10 +364,35 @@ def featurize_many(self, entries, ignore_errors=False, return_errors=False,
                                            return_errors=return_errors,
                                            pbar=pbar)
             with Pool(self.n_jobs) as p:
-                func = partial(self.featurize_wrapper,
+                func = partial(self.featurize_imap_wrapper,
                                return_errors=return_errors,
                                ignore_errors=ignore_errors)
-                return p.map(func, entries, chunksize=self.chunksize)
+                entries = list(enumerate(entries))  # turns entries into [(0, entry_1), (1, entry_2), (2, entry_3), ... ]
+                results = []
+                #set chunksize if self.chunksize=None as imap doesnt support chunksize=None
+                if self.chunksize is None:
+                    chunksize, extra = divmod(len(entries), self.n_jobs * 4)
+                    if extra:
+                        chunksize += 1
+                else:
+                    chunksize = self.chunksize
+                for i, result in tqdm_func(
+                        p.imap_unordered(func, entries, chunksize=chunksize), total=len(entries)):
+                    results.append(result)
+                return results
+
+    def featurize_imap_wrapper(self, data, return_errors=False, ignore_errors=False):
```

Comment:
Hi Karan, please can you look into this?

Reply:
I tried a lambda and a local function and both tries failed because they couldn't be pickled.
lambda: AttributeError: Can't pickle local object 'BaseFeaturizer.featurize_many.<locals>.<lambda>'
local function: AttributeError: Can't pickle local object 'BaseFeaturizer.featurize_many.<locals>.func'

Reply:
I'm not sure if this can be done with a local function (lambda or otherwise). It might be possible to use some combination of `partial` and `partial_method` from [functools](https://docs.python.org/3/library/functools.html), but that sounds tricky. I'd be fine with leaving this as a class method, but think it should be a private function.